### PR TITLE
Switch iOS library namespaces to the new ones

### LINKED ISSUE #99: 'Analytics/SEGAnalytics.h' file not found
After the upgrade to `3.4.1` I am now getting the following error:

`pub-cache/hosted/pub.dartlang.org/flutter_segment-3.4.1/ios/Classes/FlutterSegmentPlugin.m:2:9: fatal error: 'Analytics/SEGAnalytics.h' file not found`

I've tried the usual troubleshooting steps such as `flutter clean`, deleting derived data etc. Any idea why this would be happening?

### PR DESCRIPTION
**Pull request by esarbanis**
Fixes #99

### CODE REVIEW

File: example/ios/Podfile [hunk 1]
```diff
@@ -28,6 +28,8 @@ require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelpe
 flutter_ios_podfile_setup
 
 target 'Runner' do
+  use_frameworks!
```
Comment (esarbanis) [inline example/ios/Podfile:31]: IDK if I should include this as instruction in documentation. If this is not present then it will break the build.

**Comment by danielgomezrico** [comment]
@esarbanis why is this change required? can you give a little of context into the PR description? I don't understand why those namespaces "are broken" but still compiles fine on CI

**Comment by danielgomezrico** [comment]
Or what can we do in order to have this check on CI? with that we can avoid that this happens again in the future

File: example/ios/Podfile (hunk 1 above)
Comment (esarbanis) [reply example/ios/Podfile:31]: @danielgomezrico if you remove this, the pipelines will break, but this is a common function call if you use an extension in iOS.

**Comment by esarbanis** [comment]
@danielgomezrico I have opened the discussion [here](https://github.com/claimsforce-gmbh/flutter-segment/pull/100#discussion_r752617590)

**Comment by danielgomezrico** [comment]
@esarbanis but I was asking for the imports change https://github.com/claimsforce-gmbh/flutter-segment/pull/100/files#diff-8973d8942a9f1dfab770beb4da7819dad89a71409620dfad5278128b0a252ad4R2-R4, this ones, why the CI is working fine without changing those imports?

**Comment by esarbanis** [comment]
@danielgomezrico those imports are related to `use_frameworks!` in the `Podfile` target. If you remove it the pipelines will break as [they did](https://github.com/claimsforce-gmbh/flutter-segment/runs/4187164910?check_suite_focus=true) when I first pushed this branch (without use_frameworks!)

**Comment by danielgomezrico** [comment]
Oh, ok, got it, LGTM

**Review by danielgomezrico — APPROVED** [review]
LGTM

**Comment by esarbanis** [comment]
@danielgomezrico mind you that if we merge this it will break the apps that do not have `use_frameworks!`, so maybe we should mention it in the documentation as breaking change. WDYT?

**Comment by danielgomezrico** [comment]
🤔  got it.

Theres one extra thing, if we check the current build it is not failing (check this PR https://github.com/claimsforce-gmbh/flutter-segment/pull/101) then I still dont understand why this fails, that may say that the CI is broken or that theres something else related to this, @esarbanis  WDYT?

**Comment by esarbanis** [comment]
CI is building the example. Initially, the example didn't have `use_frameworks!` so the pipeline failed. It was also failing in my local environment when I tried it. When I put `use_frameworks!` in the example both pipelines and my local environment were built with no errors.

**Comment by MaiKaY** [comment]
> so maybe we should mention it in the documentation as breaking change. WDYT?

@esarbanis 
I added a note in the [README](https://github.com/claimsforce-gmbh/flutter-segment#important-note-for-ios-usage)

**Comment by MaiKaY** [comment]
Released in `3.5.0`

**Pull request merged**